awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -440,53 +459,113 @@ def delete_if_needed(self, existing_item, handle_response=True, on_delete=None):
+            endpoint = item.get('related', {}).get(post['endpoint_reference'], None)
+            if endpoint:
+                # First get the existing data from the endpoint
+                existing_data = self.get_endpoint(endpoint)
+
+                # If the endpont data is identical, just continue on to the next item
+                if existing_data['json'] == post['data']:
+                    continue
+
+                with open('/tmp/john', 'w') as f:
+                    import json
+                    f.write(json.dumps(post['data']))
+
+                # If this item no longer has data, lets remove it
+                if post['data'] in ('', [], {}):
+                    response = self.delete_endpoint(endpoint)
+                    # Tower job template requires a 200
+                    # Left as an array as for future additional delete returns
+                    if response['status_code'] in [200]:
+                        self.json_output['deleted_additional_post'] = True
+                        self.json_output['changed'] = True
                     else:
-                        self.fail_json(msg="Unable to delete {0} {1}: {2}".format(item_type, item_name, response['json']))
+                        self.fail_json(msg="Unable to delete to {0} for {1}: {2}".format(endpoint, item_type, self.extract_errors_from_response(response)), **{'response': response})
                 else:
-                    self.fail_json(msg="Unable to delete {0} {1}: {2}".format(item_type, item_name, response['status_code']))
-        else:
-            if not handle_response:
-                return None
+                    response = self.post_endpoint(endpoint, **{'data': post['data']})
+                    # Tower job template requires a 200
+                    # Left as an array as for future additional posts returns
+                    if response['status_code'] in [200]:
+                        self.json_output['added_additional_post'] = True
+                        self.json_output['changed'] = True
+                    else:
+                        self.fail_json(msg="Unable to post to {0} for {1}: {2}".format(endpoint, item_type, self.extract_errors_from_response(response)), **{'response': response})
             else:
-                self.exit_json(**self.json_output)
+                self.fail_json(msg="Addition post type of {0} is not valid for {1}".format(association_type, item_type))
+
+    def process_associations(self, item, item_type, associations):
+        # Process any associations with this item
+        if associations is not None:
+            for association_type in associations:
+                if item.get('related', {}).get(association_type, None):
+                    self.modify_associations(item['related'][association_type], associations[association_type])
+                else:
+                    self.fail_json(msg="Association type of {0} is not valid for {1}".format(association_type, item_type))
```

Comment:
this stuff should get cleaned up after rebasing

Reply:
I feel like we have a little bit of a mess on our hands with all of the things flying around @beeankha and I are going to try and cleanup and rebase on Monday.

Reply:
The existing item, `item` in this case will be `None` in some cases. This comes from the case where the object is created, and has associations, and the fix I applied to that doesn't look like it will work here. Some test coverage for this was introduced in https://github.com/ansible/awx/pull/6369. What changed with the processing of new items in this PR, and how is it expected that the case of create-and-associate will work with the call patterns as they were changed here?